### ISSUE TITLE
[Bug]: Validation not working in Jokes Tutorial

### ISSUE THREAD

**Issue by RichardHpa** [remix-run/remix#1852]
### Which Remix packages are impacted?

- [X] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.1.3

### What version of Node are you using? Minimum supported version is 14.

16.13.0

### Steps to Reproduce

Follow the Remix Jokes tutorial on a new instance of Remix (or update them to the current version). Go to the add new jokes route and try and submit the form without entering any input fields. You wont receive any validation errors.

### Expected Behavior

I am expecting to have the validation errors appear with the input fields.

<img width="1142" alt="Screen Shot 2022-02-09 at 10 54 24 PM" src="https://user-images.githubusercontent.com/26518752/153172765-3a22fe32-c5a1-45a9-8636-c4aa10dcd43c.png">
 
I also did a `console.log(actionData)` in the `<NewJokeRoute/>` and I expect to see an object with field errors included.

<img width="463" alt="Screen Shot 2022-02-09 at 10 56 53 PM" src="https://user-images.githubusercontent.com/26518752/153173432-4cdb1116-9180-43a6-a377-cf79fe189b90.png">

These snapshots were taken when I downgraded my remix version to be `v1.0.4` which the original jokes app was created on.

### Actual Behavior

Rather than showing the errors nothing appears at all
<img width="1133" alt="Screen Shot 2022-02-09 at 10 54 44 PM" src="https://user-images.githubusercontent.com/26518752/153173983-a7dea48b-0c2e-413c-b974-0b639779f5d1.png">

And when I include `console.log(actionData)` in the `<NewJokeRoute/>` I only see the `FormError` value.

<img width="488" alt="Screen Shot 2022-02-09 at 10 57 05 PM" src="https://user-images.githubusercontent.com/26518752/153174308-463d4ef4-2b86-4384-a353-2945ac5eb8cf.png">

**Comment by mkrtchian** [comment]
I also had the problem when I followed the tutorial with version 1.1.3. I just reproduced it with [the jokes example code](https://github.com/remix-run/remix/tree/main/examples/jokes), and indeed with Remix 1.0.6 it works as before, and from 1.1.0 it has the new behavior.

I think the change was introduced in MR #743 where the default Request body is set to `null` and no more to empty string.

In the jokes app code itself it is because of [this check](https://github.com/remix-run/remix/blob/main/examples/jokes/app/routes/jokes/new.tsx#L57) returning a status with the `formError` and not with the `fieldErrors`:
```tsx
  if (typeof name !== "string" || typeof content !== "string") {
    return badRequest({ formError: `Form not submitted correctly.` });
  }
```
We can replace it with a check that lets `null` pass, and handle it later in the individual field validation functions that will return messages in `fieldErrors` so that the error is displayed as before:
```tsx
  if (
    (typeof name !== "string" && name !== null) ||
    (typeof content !== "string" && content !== null)
  ) {
    return badRequest({
      formError: `Form not submitted correctly.`,
    });
  }
```

**Comment by RichardHpa** [comment]
What I ended up doing was adding an or when you get the variables from `form.get()`. This meant I didn't have to change the types for the validator.

```javascript
  const name = form.get('name') || '';
  const content = form.get('content') || '';
```

But even with this workaround it would be great to try and figure out what the change was within remix and get it resolved.

**Comment by machour** [comment]
Fixed in #1869
